File: backend/src/main/java/br/com/visitsafe/dto/condominium/CondominiumResponseDTO.java
package br.com.visitsafe.dto.condominium;

import br.com.visitsafe.model.enums.CondominiumTypeEnum;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.UUID;

@Data
@AllArgsConstructor
public class CondominiumResponseDTO {
    private UUID id;
    private String name;
    private String cnpj;
    private CondominiumTypeEnum condominiumType;
}

File: frontend/src/api/models/CondominiumResponseDTO.ts
export const condominiumResponseDTOCondominiumTypeEnum = {
  BUSINESS: 'BUSINESS',
  RESIDENTIAL: 'RESIDENTIAL',
} as const

export type CondominiumResponseDTOCondominiumTypeEnum =
  (typeof condominiumResponseDTOCondominiumTypeEnum)[keyof typeof condominiumResponseDTOCondominiumTypeEnum]

export type CondominiumResponseDTO = {
  /**
   * @type string | undefined, uuid
   */
  id?: string
  /**
   * @type string | undefined
   */
  name?: string
  /**
   * @type string | undefined
   */
  cnpj?: string
  /**
   * @type string | undefined
   */
  condominiumType?: CondominiumResponseDTOCondominiumTypeEnum
}